docs/configure/sidebar-and-urls.md

```diff
@@ -139,50 +139,55 @@ When Storybook generates the titles for all matching stories, they'll retain the
+| `title`           | If the `Meta` has a `title` attribute, it will be used as the title.                                                           | `export default { title: 'My Title' }`                        |
+| `component`       | If the `Meta` has a `component` attribute, it will be used as the title.                                                       | `export default { component: MyComponent }`                   |
+| \*\* File Name    | used if `component` does not exist and `title` does not exist                                                                  |
+| `name`            | If the Story definition has a `name` attribute, it will be used as the title. This was formerly `storyName`                    | `export const Primary: Story = { name: 'My Title' }`          |
+| \*\* Story Export | used if `name` does not exist                                                                                                  |
+| `title`           | If the `Meta` tag has a `title` attribute, it will be used as the title.                                                       | `<Meta title="My Title" />`                                   |
+| `of`              | If the `Meta` tag has a `of` attribute, it will be used to retrieve the title from the referenced story file.                  | `<Meta of={MyComponentStories} />`                            |
+| `name`            | If the `Story` tag has a `name` attribute, it will be used as the title.                                                       | `<Story name="My Story Title" />`                             |
 
 <!-- prettier-ignore-start -->
 
 <CodeSnippets
   paths={[
-    'common/storybook-main-csf-indexer.js.mdx',
+    'common/csf-3-example-title.ts.mdx',
+    'common/csf-3-example-title.mdx.mdx',
   ]}
 />
 
 <!-- prettier-ignore-end -->
 
-Another example listed below for indexing `.md` & `.html` files which is already implemented by one of our community addons [`Markdown Docs`](https://storybook.js.org/addons/@sheriffmoose/storybook-md/).
+Storybook provides capability for customizing the title generation process. This is done by providing a `makeTitle` function to the `loadCsf` method in the `storyIndexer`.
 
 <!-- prettier-ignore-start -->
 
 <CodeSnippets
   paths={[
-    'common/storybook-main-md-html-indexer.js.mdx',
+    'common/storybook-main-story-indexer-main.js.mdx'
   ]}
 />
 
 <!-- prettier-ignore-end -->
+
+A good example of this is the [`@storybook-extras/markdown` ](https://storybook.js.org/addons/@storybook-extras/markdown/) addon which uses the `makeTitle` to generate the title for the `DocsPage` component. It uses the `makeTitle` to generate the title based on few rules related to the `.md` & `.html` files: `\<title\>` tag, `\<meta\>` tag, `{/**/}` markdown comment, custom file names or even a predfined list of titles in the `main.js` file.
```

Comment:
@sheriffMoose, once again we have a bit of room for improvement here, I was thinking of adjusting this section to something more along the lines of:
```md
### Processing custom titles

Out of the box, Storybook automatically infers the story's title based on a set of patterns, including, for example, the file's physical location. If you're working on creating a custom story indexer and you want to handle the titles based on your set of rules, you can adjust it and provide a `makeTitle` function inside the `loadCsf` function. Below is a condensed table and examples of how these patterns work and how they relate to stories.

| Pattern        | Description                                                                                                                                    |
| -------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
| `titlePrefix`  | Prefixes the indexed story title based on the [configuration](./overview.md#with-a-configuration-object).                                      |
| `title`        | Requires CSF and MDX.<br/>Infers the story title based on the information defined in the metadata.                                             |
| `name`         | Requires CSF and MDX.<br/> Overrides the story title based on the provided [name](../writing-stories/introduction.md#rename-stories) property. |
| `component`    | Requires CSF.<br/> Generates the title based on the component property defined in the metadata.                                                |
| `of`           | Requires MDX.<br/> Retrieves the title based on the referenced story file.                                                                     |
| `story export` | Requires CSF. <br/> Defines the title based on the [named story export](../api/csf.md#named-story-exports).                                    |
| `filename`     | Enabled with custom indexers.<br/> Defines the story title based on the filename parsed with the indexer.                                      |

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'common/csf-3-example-title.ts.mdx',
    'common/csf-3-example-title.mdx.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'common/storybook-main-story-indexer-main.js.mdx'
  ]}
/>

<!-- prettier-ignore-end -->

<div class="aside">

💡 Story indexers are a great way to get non-developers playing with your components and stories. They can do much more than documented here; we encourage you to check out the [`@storybook-extras/markdown` ](https://storybook.js.org/addons/@storybook-extras/markdown/) to learn more about these techniques.

</div>
```

You'll need to make some adjustments to the examples to match the table structure. What do you think?

Reply:
thanks for the remarks :) this is helpful insight for me to get better at these documentations